src/OrleansTelemetryConsumers.Counters/PerfCounterTelemetryConsumer.cs

```diff
@@ -0,0 +1,224 @@
+﻿using Orleans;
+using Orleans.Runtime;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+
+namespace OrleansTelemetryConsumers.Counters
+{
+    public class PerfCounterTelemetryConsumer : IMetricTelemetryConsumer
```

Comment:
Please rename to something like OrleansPerfCountersTelemetryConsumer.  This impl is very specific to Orleans runtime performance counters.  A more general metric telemetry consumer can be added later, but this one is specific and the name should communicate this.

For initial impl, I think it's ok to limit this to Orleans runtime performance counters, but